graspologic/plot/plot.py

```diff
@@ -671,6 +676,228 @@ def pairplot(
     return pairs
 
 
+def _plot_ellipse_and_data(
+    data, X, j, k, means, covariances, ax, label_palette, cluster_palette, alpha
+):
+    r"""
+    plot_ellipse makes a scatter plot from the two dimensions j,k where j
+    corresponds to x-axis
+    and k corresponds to the y-axis onto the axis that is ax. plot_ellipse then
+    applies a gmm ellipse onto the scatterplot
+    using the data from Y_(which is stored in data["clusters"]),
+    means, covariances.
+
+    Parameters
+    ----------
+    X : array-like, shape (n_samples, n_features)
+        Input data.
+    j : int
+        column index of feature of interest from X which will be the x-axis data.
+    k : int
+        column index of feature of interest from X which will be the y-axis data.
+    means : array-like, shape (n_components, n_features)
+        Estimated means from gmm
+    covariances : array-like, shape (with 'full') (n_components, n_features, n_features)
+        estimated covariances from gmm
+    ax : axis object
+        The location where plot_ellipse will plot
+    label_palette : dict, optional, default: dictionary using 'Set1'
+    cluster_palette : dict, optional, default: dictionary using 'Set1'
+    alpha : float, optional, default: 0.7
+        Opacity value of plotter markers between 0 and 1
+
+    References
+    ----------
+    .. [1] Weiss, R., Varoquaux, G.(2011). Scikit-learn: Machine Learning in PythonJournal
+        of Machine Learning Research, 12, 2825–2830.
+    """
+    sns.scatterplot(
+        data=data, x=X[:, j], y=X[:, k], ax=ax, hue="labels", palette=label_palette
+    )
+    for i, (mean, covar) in enumerate(zip(means, covariances)):
+        v, w = linalg.eigh(covar)
+        v = 2.0 * np.sqrt(2.0) * np.sqrt(v)
+        u = w[0] / linalg.norm(w[0])
+        # Plot an ellipse to show the Gaussian component
+        angle = np.arctan(u[1] / u[0])
+        angle = 180.0 * angle / np.pi
+        ell = mpl.patches.Ellipse(
+            [mean[j], mean[k]],
+            v[0],
+            v[1],
+            180.0 + angle,
+            color=cluster_palette[i],
+        )
+        ell.set_clip_box(ax.bbox)
+        ell.set_alpha(alpha)
+        ax.add_artist(ell)
+        # removes tick marks from off diagonal graphs
+
+        ax.set_xticks([]), ax.set_yticks([])
+        ax.set_ylabel(k), ax.set_xlabel(k)
+        ax.legend().remove()
+
```

Comment:
`ax.set(a, b, ..., n)` would be a little cleaner

Reply:
https://github.com/microsoft/graspologic/pull/541/commits/3d4155a9c13818a75c936fff5d269201a0b4491d yep fixed!